### DISCUSSION TITLE
Cannot import the accuracy, f1 score and accuracy from the pytorch lightning metric library

### DISCUSSION THREAD

**Discussion by shubheshswain91** [Lightning-AI/pytorch-lightning#10253] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am getting the following import error:

```
ImportError: cannot import name 'r2score' from 'torchmetrics.functional' (/usr/local/lib/python3.7/dist-packages/torchmetrics/functional/__init__.py)
```

I have installed the pytorch lightning library as seen by the below installation log:
```
Requirement already satisfied: pytorch-lightning in /usr/local/lib/python3.7/dist-packages (1.3.0)
Requirement already satisfied: tensorboard!=2.5.0,>=2.2.0 in /usr/local/lib/python3.7/dist-packages (from pytorch-lightning) (2.6.0)
Requirement already satisfied: pyDeprecate==0.3.0 in /usr/local/lib/python3.7/dist-packages (from pytorch-lightning) (0.3.0)
Requirement already satisfied: torch>=1.4 in /usr/local/lib/python3.7/dist-packages (from pytorch-lightning) (1.9.0+cu111)
Requirement already satisfied: torchmetrics>=0.2.0 in /usr/local/lib/python3.7/dist-packages (from pytorch-lightning) (0.6.0)
```

What is the alternate way to import the auroc, accuracy, and F1 score?

**Comment by rohitgr7** [comment]
seems to be torchmetrics related issue. can you open a discussion here: https://github.com/PyTorchLightning/metrics instead?

**Comment by akihironitta** [comment]
@shubheshswain91 Hi, the error happens when using `pytorch_lightning<=1.4` with `torchmetrics>=0.6`.

If you're using:
```python
# don't use this
from pytorch_lightning.metrics import ...
```
, it is deprecated and will be removed in v1.5. Instead, please use:
```python
# use this
from torchmetrics import ...
```

<details>
<summary>or you could...</summary>

downgrade it to `torchmetrics<0.6` and use:

```python
from pytorch_lightning.metrics import ...
```

but not recommended.
</details>

**Reply by Fang-git0** [reply]
Error:
ImportError: cannot import name 'r2score' from 'torchmetrics.functional'

My version：
python==3.7
torch 1.8.0+cu111
torchmetrics 0.6.0
torchtext 0.9.0
pytorch-lightning 1.4.9

change version to torchmetrics==0.5.1
errors：
ImportError: cannot import name 'Batch' from 'torchtext.data'

change version to pytorch-lightning==1.3.8
error：ImportError: cannot import name 'f1_score' from 'pytorch_lightning.metrics.functional'

**Reply by Fang-git0** [reply]
I have the same problem...
How to solve it...

**Reply by rohitgr7** [reply]
this has nothing to do with pytorch-lightning version. On `torchmetrics==0.6.0`, its `from torchmetrics.functional import r2_score`.

ref: https://torchmetrics.readthedocs.io/en/v0.6.0/references/functional.html#r2-score-func

**Reply by Fang-git0** [reply]
Thank you so much. Solved it.

**Comment by itzortzis** [comment]
As @Fang-git0  suggested, I changed to torchmetrics version 0.5.1 and it works like a charm. Thanks!
